web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileBlock.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dbcdk_community\Plugin\Block\ProfileBlock.
+ */
+
+namespace Drupal\dbcdk_community\Plugin\Block;
+
+use Drupal\dbcdk_community\CommunityTraits;
+use DBCDK\CommunityServices\Model\Profile;
+use DBCDK\CommunityServices\Api\ProfileApi;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Url;
+use Drupal\Component\Utility\Xss;
+
+/**
+ * Provides a 'ProfileBlock' block.
+ *
+ * This block provides a display of a single Community Service Profile.
+ *
+ * @Block(
+ *   id = "dbcdk_community_profile_block",
+ *   admin_label = @Translation("DBCDK Community Profile"),
+ *   context = {
+ *     "username" = @ContextDefinition("string")
+ *   }
+ * )
+ */
+class ProfileBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  use CommunityTraits;
+
+  /**
+   * The DBCDK Community Service Profile API.
+   *
+   * @var ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * Creates a Profiles Block instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
+   *   The DBCDK Community Service Profile API.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ProfileApi $profile_api) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->profileApi = $profile_api;
```

Comment:
Would it be possible to load the Profile in `create` method and just having a Profile as a variable?